src/common/hooks/useEditForm.js

```diff
@@ -4,14 +4,30 @@ import { yupResolver } from '@hookform/resolvers/yup'
 
 // FPCC
 import { isUUID } from 'common/utils/stringHelpers'
+import { useSiteStore } from 'context/SiteContext'
+import useAuthCheck from 'common/hooks/useAuthCheck'
+import { TEAM } from 'common/constants'
 
 function useEditForm({ defaultValues, validator, dataToEdit }) {
   const isEditMode = isUUID(dataToEdit?.id)
+  const { site } = useSiteStore()
+  const { checkIfAssistant } = useAuthCheck()
+  const isAssistant = checkIfAssistant()
 
-  const data = useMemo(
-    () => (dataToEdit?.id ? dataToEdit : null),
-    [dataToEdit?.id],
-  )
+  const data = useMemo(() => (dataToEdit?.id ? dataToEdit : null), [dataToEdit])
```

Comment:
useEditForm is used by 33 different forms in our app., many of which don't include visibility. I think it would be best if we consider another solution that doesn't have such a wide reaching impact.